ssl/quic/quic_obj.c

```diff
@@ -80,10 +81,53 @@ static int obj_update_cache(QUIC_OBJ *obj)
 
 SSL_CONNECTION *ossl_quic_obj_get0_handshake_layer(QUIC_OBJ *obj)
 {
-    assert(obj->init_done);
+    assert(obj != NULL && obj->init_done);
 
-    if (obj == NULL || obj->ssl.type != SSL_TYPE_QUIC_CONNECTION)
+    if (obj->ssl.type != SSL_TYPE_QUIC_CONNECTION)
         return NULL;
 
     return SSL_CONNECTION_FROM_SSL_ONLY(((QUIC_CONNECTION *)obj)->tls);
 }
+
+/* (Returns a cached result.) */
+int ossl_quic_obj_can_support_blocking(const QUIC_OBJ *obj)
+{
+    QUIC_REACTOR *rtor;
+
+    assert(obj != NULL);
+    rtor = ossl_quic_obj_get0_reactor(obj);
+
+    return ossl_quic_reactor_can_poll_r(rtor)
+        || ossl_quic_reactor_can_poll_w(rtor);
+}
+
+int ossl_quic_obj_desires_blocking(const QUIC_OBJ *obj)
+{
+    unsigned int req_blocking_mode;
+
+    assert(obj != NULL);
+    for (; (req_blocking_mode = obj->req_blocking_mode)
+            == QUIC_BLOCKING_MODE_INHERIT && obj->parent_obj != NULL;
```

Comment:
Nit; I prefer splitting for readability at `&&` even though the line will be longer than 80 chars.